app/components/polaris/page_component.rb

```diff
@@ -44,8 +46,8 @@ def header_arguments
         tag: "div",
         classes: class_names(
           "Polaris-Page-Header--mobileView",
-          "Polaris-Page-Header--mediumTitle": @title.present? && @title.length <= LONG_TITLE,
-          "Polaris-Page-Header--longTitle": @title.present? && @title.length > LONG_TITLE,
+          "Polaris-Page-Header--mediumTitle": @title.present? && (@force_medium_title || @title.length <= LONG_TITLE),
```

Comment:
@yshmarov Maybe stripping HTML from `@title` before checking length will be easier?
https://api.rubyonrails.org/classes/ActionView/Helpers/SanitizeHelper.html#method-i-strip_tags

Reply:
this is indeed much better. I've updated

Reply:
Looks good! 👍 Please check tests and we can merge it